packages/contracts/test/contracts/OVM/accounts/OVM_ECDSAContractAccount.spec.ts

```diff
@@ -190,4 +192,23 @@ describe('OVM_ECDSAContractAccount', () => {
       )
     })
   })
+  describe('isValidSignature()', () => {
+    const message = '0x42'
+    const messageHash = ethers.utils.hashMessage(message)
+    it(`should revert for a malformed signature`, async () => {
+      await expect(
+        OVM_ECDSAContractAccount.isValidSignature(messageHash, '0xdeadbeef')
+      ).to.be.revertedWith('ECDSA: invalid signature length')
+    })
+    it(`should return 0 for an invalid signature`, async () => {
+      const signature = await wallet.signMessage(message)
+      const bytes = await OVM_ECDSAContractAccount.isValidSignature(
+        messageHash,
+        signature
+      )
+      expect(bytes).to.equal('0x00000000')
+    })
+    // NOTE: There is no good way to unit test verifying a valid signature
+    // An integration test exists testing this instead
```

Comment:
@smartcontracts set up the `Lib_ExecutionManagerWrapper.ovmADDRESS` to execute in a separate delegated call frame so that it can be `smock`ed, see "`smocked.ovmADDRESS`" elsewhere in these tests.  This is why we otherwise use `ovmADDRESS` in the contract account instead of `address(this)`.  Alternatively in this case we could use `await smockit('OVM_ECDSAContractAccount', {address: wallet.address})` to make the `address(this)` work in this case IIRC.  Either way, I think the integration test we have is probably fine for now

Reply:
can't use `Lib_ExecutionManagerWrapper.ovmADDRESS` because we need to keep `isValidSignature` as a view function. Going to try out that smockit approach

Reply:
:/ smockit didn't work for this, going to just keep my integration test